### PR TITLE
Fix elastic formatting rule with comments

### LINKED ISSUE #1947: Formatting adds unexpected extra line break
In the code below, the resulting string contains two blank lines above "example comment" after formatting. I would expect that the single blank line is preserved. Repro for CTP6.

    using System.Linq;
    using Microsoft.CodeAnalysis;
    using Microsoft.CodeAnalysis.CSharp.Syntax;
    using Microsoft.CodeAnalysis.Editing;
    using Microsoft.CodeAnalysis.Formatting;
    using Microsoft.CodeAnalysis.Text;
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    namespace RoslynTestCtp6
    {
        [TestClass]
        public class RoslynFormattingBug
        {
            [TestMethod]
            public void Formatter_inserts_additional_line_break()
            {
                const string source = @"

                        public class C
                        {
                            public string f1;
                    
                            // example comment
                            public string f2;
                        }

                        public class SomeAttribute : System.Attribute { }
                    ";

                var projectId = ProjectId.CreateNewId();
                var solution =
                    new AdhocWorkspace().CurrentSolution.AddProject(projectId, "test", "test", LanguageNames.CSharp)
                        .AddMetadataReference(projectId, MetadataReference.CreateFromAssembly(typeof (object).Assembly));

                var documentId = DocumentId.CreateNewId(projectId, "repro");
                solution = solution.AddDocument(documentId, "repro", SourceText.From(source));

                var document = solution.Projects.First().Documents.First();
                var compilation = solution.Projects.First().GetCompilationAsync().Result;

                var typeSymbol = compilation.GetTypeByMetadataName("C");
                var fieldSymbol = typeSymbol.GetMembers("f2").First();

                var fieldSyntax = fieldSymbol.DeclaringSyntaxReferences.First().GetSyntax();
                var varSyntax = (VariableDeclarationSyntax) fieldSyntax.Parent;

                var attribute = compilation.GetTypeByMetadataName("SomeAttribute");

                var editor = DocumentEditor.CreateAsync(document).Result;
                SyntaxNode attributeSyntax =
                    editor.Generator.Attribute(editor.Generator.TypeExpression(attribute))
                        .WithAdditionalAnnotations(Formatter.Annotation);
                editor.AddAttribute(varSyntax, attributeSyntax);

                var newRoot = editor.GetChangedRoot();
                var newDocument = editor.GetChangedDocument();

                SyntaxNode rootFormatted = Formatter.Format(newRoot, newDocument.Project.Solution.Workspace);
                string result = rootFormatted.ToString();

                // result now contains:
                /*
                public class C
                {
                    public string f1;

                    // example comment
                    [global::SomeAttribute]
                    public string f2;
                }

                public class SomeAttribute : System.Attribute { }
                */
            }
        }
    }

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #1947 

Change the elastic formatting rule to take into account the existing number of blank lines before a comment (between two declarations) when determining the correct number of blank lines that should appear between the two declarations.

@pilchie @mavasani @heejaechang  please review

### CODE REVIEW

**Comment by heejaechang** [comment]
:+1:

File: src/Workspaces/CSharp/Portable/Formatting/Rules/ElasticTriviaFormattingRule.cs [hunk 1]
```diff
@@ -189,7 +189,9 @@ private AdjustNewLinesOperation GetAdjustNewLinesOperationBetweenMembers(SyntaxT
             {
                 // the first one is a comment, add two more lines than existing number of lines
```
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/ElasticTriviaFormattingRule.cs]: Update the comment here?

File: src/Workspaces/CSharp/Portable/Formatting/Rules/ElasticTriviaFormattingRule.cs [hunk 2]
```diff
@@ -444,7 +446,7 @@ private bool IsWhitespace(SyntaxTrivia trivia)
                 || trivia.Kind() == SyntaxKind.EndOfLineTrivia;
         }
 
-        private int GetNumberOfLines(SyntaxTriviaList triviaList)
+        private int GetNumberOfLines(IEnumerable<SyntaxTrivia> triviaList)
```
Comment (Pilchie) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/ElasticTriviaFormattingRule.cs]: I almost asked why make this change when you now force boxing the `SyntaxTriviaList`, but then I looked at the implementation and realized that it probably doesn't matter...

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**